apps/weather_status/src/App.vue

```diff
@@ -0,0 +1,503 @@
+<!--
+  - @copyright Copyright (c) 2020 Julien Veyssier <eneiluj@posteo.net>
+  - @author Julien Veyssier <eneiluj@posteo.net>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<li :class="{ inline }">
+		<div id="weather-status-menu-item">
+			<Actions
+				class="weather-status-menu-item__subheader"
+				:default-icon="weatherIcon"
+				:menu-title="visibleMessage">
+				<ActionLink v-if="address && !errorMessage"
+					icon="icon-address"
+					target="_blank"
+					:href="weatherLinkTarget"
+					:close-after-click="true">
+					{{ locationText }}
+				</ActionLink>
+				<ActionSeparator v-if="address && !errorMessage" />
+				<ActionButton
+					icon="icon-crosshair"
+					:close-after-click="true"
+					@click="onBrowserLocationClick">
+					{{ t('weather_status', 'Detect location') }}
+				</ActionButton>
+				<ActionInput
+					ref="addressInput"
+					:disabled="false"
+					icon="icon-rename"
+					type="text"
+					value=""
+					@submit="onAddressSubmit">
+					{{ t('weather_status', 'Set custom address') }}
+				</ActionInput>
+			</Actions>
+		</div>
+	</li>
+</template>
+
+<script>
+import { showError } from '@nextcloud/dialogs'
+import moment from '@nextcloud/moment'
+import { getLocale } from '@nextcloud/l10n'
+import Actions from '@nextcloud/vue/dist/Components/Actions'
+import ActionButton from '@nextcloud/vue/dist/Components/ActionButton'
+import ActionInput from '@nextcloud/vue/dist/Components/ActionInput'
+import ActionLink from '@nextcloud/vue/dist/Components/ActionLink'
+import ActionSeparator from '@nextcloud/vue/dist/Components/ActionSeparator'
+import * as network from './services/weatherStatusService'
+
+const MODE_BROWSER_LOCATION = 1
+const MODE_MANUAL_LOCATION = 2
+const weatherOptions = {
+	clearsky_day: {
+		icon: 'icon-clearsky-day',
+		text: t('weather_status', 'Clear sky'),
```

Comment:
You actually "string concatenate" quite hard here: `71.4 °F Clear sky at 2:00 AM`

As per https://docs.nextcloud.com/server/19/developer_manual/app/view/l10n.html#improving-your-translations that is quite bad. It would be better to have this strings as:

```suggestion
		text: t('weather_status', '{temperature} Clear sky at {time}'),
```

Than translators can make sure it matches their language. and you either string-replace the values in, or actually provide them as parameters to the translation.